core/controllers/acl_decorators.py

```diff
@@ -303,6 +303,68 @@ def test_can_edit(self, collection_id, **kwargs):
     return test_can_edit
 
 
+def can_edit_suggestion(handler):
+    """Decorator to check whether the user can edit given suggestion.
+
+    Args:
+        handler: function. The function to be decorated.
+
+    Returns:
+        function. The newly decorated function that now also checks if
+            a user has permission to edit a given suggestion.
+    """
+
+    def test_can_edit_suggestion(self, **kwargs):
```

Comment:
A lot of the code here can be simplified, I believe, after simplifying the check_can_edit_suggestion function.

Reply:
done PTAL @nithusha21